### PR TITLE
`Programming exercises`: Fix clickable checkbox on detail view

### PR DESCRIPTION
**Pull request by Strohgelaender**
#### General
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.

### Motivation and Context
The pe details page is only for viewing the exercise settings, you should not be able to change them on this page. However, the checkbox for showing test case names to students was still editable here.

### Description
I added the missing `disabled` property.

### Steps for Testing
View the PE details page, see that the checkbox is not clickable.
Edit the PE, view that the checkbox is clickable here.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots
old:
grafik

new:
grafik

### CODE REVIEW

**Review by b-fein — APPROVED** [review]
Looks good.

(optional) Another small inconsistency visible in light mode that could be fixed:
image


The text for the last item is greyed out due to line 178:
```
[ngStyle]="!exercise.exampleSolutionPublicationDate ? { color: 'grey' } : {}"
```

**Comment by Strohgelaender** [comment]
@b-fein I think this is actually correct. The text is grayed out since it's not possible to activate this option if no example solution date is set. However, I don't think this is a difference worth highlighting during the details view, so the text now only turns grey while editing the exercise.

File: src/main/webapp/app/exercises/programming/shared/lifecycle/programming-exercise-lifecycle.component.html [hunk 1]
```diff
@@ -25,7 +25,7 @@
 
             <div class="test-schedule-date px-1">
                 <div class="test-schedule-date-title test-schedule-date-title-small-button">
-                    <span class="fw-bold" jhiTranslate="artemisApp.programmingExercise.timeline.automaticTests"> Automatic Tests </span>
+                    <span jhiTranslate="artemisApp.programmingExercise.timeline.automaticTests"> Automatic Tests </span>
```
Comment (Strohgelaender) [inline src/main/webapp/app/exercises/programming/shared/lifecycle/programming-exercise-lifecycle.component.html:28]: Fixes this inconsistent bold text: 
grafik

**Review by terlan98 — APPROVED** [review]
Tested on TS5. Works as expected, and the code looks good 👍

**Review by b-fein — APPROVED** [review]
Looks good, both in code and in the browser.

**Review by basak-akan — APPROVED** [review]
code looks good 👍

**Review by DominikRemo — APPROVED** [review]

**Pull request merged**